Responses propagate to other tests unless responses.activate is used

```python
import requests
import responses


def test_1():
    responses.add(responses.GET, 'http://example.net', json={'status': 'ok'})
    assert requests.get('http://example.net').json()['status'] == 'ok'


def test_2():
    responses.add(responses.GET, 'http://example.net', json={'status': 'fail'})
    assert requests.get('http://example.net').json()['status'] == 'fail'
```

```
========================================================= test session starts ==========================================================
platform linux -- Python 3.5.3rc1, pytest-3.0.5, py-1.4.32, pluggy-0.4.0
rootdir: /home/bertrand/test/pytest_responses_noreset, inifile: 
plugins: responses-0.1.0
collected 2 items 

test.py .F

=============================================================== FAILURES ===============================================================
________________________________________________________________ test_2 ________________________________________________________________

    @responses.activate
    def test_2():
        responses.add(responses.GET, 'http://example.net', json={'status': 'fail'})
>       assert requests.get('http://example.net').json()['status'] == 'fail'
E       assert 'ok' == 'fail'
E         - ok
E         + fail

test.py:13: AssertionError
================================================== 1 failed, 1 passed in 0.03 seconds ==================================================
```

Decorating the first test with `responses.activate` makes both tests pass.

--- a/pytest_responses.py
+++ b/pytest_responses.py
@@ -15,6 +15,7 @@ def pytest_runtest_teardown(item):
     if not item.get_marker('withoutresponses'):
         try:
             responses.stop()
+            responses.reset()
         except RuntimeError:
             # patcher was already uninstalled and responses doesnet let us
             # force maintain it

--- a/test_pytest_responses.py
+++ b/test_pytest_responses.py
@@ -22,3 +22,17 @@ def test_enabled():
         requests.get('http://responses.invalid')
 
     assert len(responses.calls) == 1
+
+
+class TestReset:
+    """
+    Ensure that the callback registered in `test_1` is cleaned up before
+    `test_2`.
+    """
+    def test_1(self):
+        responses.add(responses.GET, 'http://responses.invalid', json=1)
+        assert requests.get('http://responses.invalid').json() == 1
+
+    def test_2(self):
+        responses.add(responses.GET, 'http://responses.invalid', json=2)
+        assert requests.get('http://responses.invalid').json() == 2

